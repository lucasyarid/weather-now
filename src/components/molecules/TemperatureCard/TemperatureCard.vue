<template>
  <div class="TemperatureCard">
    <BaseCard :title="title" :bottomText="`Updated at ${updatedAt}`">
      <BaseLoading v-if="isLoading" />
      <div v-else-if="hasError" class="TemperatureCard__error">
        Something went wrong
        <button class="TemperatureCard__button" @click="$emit('tryAgain')">
          Try Again
        </button>
      </div>
      <template v-else>
        <BaseTemperature :temperature="temperature" />
        <template v-if="isExpanded" slot="footer">
          <div class="TemperatureCard__footer">
            <BaseHumidity :humidity="humidity" />
            <BasePressure :pressure="pressure" />
          </div>
        </template>
      </template>
    </BaseCard>
  </div>
</template>

<script>
import {
  BaseCard,
  BaseTemperature,
  BasePressure,
  BaseHumidity,
  BaseLoading
} from "@/components/atoms";

export default {
  name: "TemperatureCard",
  components: {
    BaseCard,
    BaseTemperature,
    BasePressure,
    BaseHumidity,
    BaseLoading
  },
  props: {
    title: {
      type: String,
      default: "loading..."
    },
    temperature: {
      type: Number,
      default: 0
    },
    humidity: {
      type: Number,
      default: 0
    },
    pressure: {
      type: Number,
      default: 0
    },
    updatedAt: {
      type: String,
      default: ""
    },
    isExpanded: {
      type: Boolean,
      default: false
    },
    isLoading: {
      type: Boolean,
      default: false
    },
    hasError: {
      type: Boolean,
      default: false
    }
  }
};
</script>

<style lang="scss" scoped>
.TemperatureCard {
  &__error {
    color: $color-red;
  }

  &__button {
    display: block;
    margin: 10px auto;
    border: 2px solid $color-darker;
    color: $color-darker;
    border-radius: 20px;
    padding: 10px 20px;
    transition: all 0.2s;
    cursor: pointer;

    &:hover {
      background-color: $color-darker;
      color: white;
    }
    &:focus,
    &:active {
      outline: none;
    }
  }

  &__footer {
    display: flex;
    justify-content: space-around;
  }
}
</style>
