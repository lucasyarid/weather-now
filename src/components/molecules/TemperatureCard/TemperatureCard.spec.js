import { shallowMount } from "@vue/test-utils";
import TemperatureCard from "./TemperatureCard";

describe("TemperatureCard", () => {
  const WRAPPER = shallowMount(TemperatureCard);

  test("Componente existe", () => {
    expect(WRAPPER.exists()).toBe(true);
  });

  test("Componente nÃ£o alterado", () => {
    expect(WRAPPER.html()).toMatchSnapshot();
  });
});
