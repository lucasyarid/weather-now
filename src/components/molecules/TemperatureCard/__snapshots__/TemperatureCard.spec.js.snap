// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemperatureCard Componente n√£o alterado 1`] = `"<div class=\\"TemperatureCard\\"><basecard-stub title=\\"loading...\\" bottomtext=\\"Updated at \\"><basetemperature-stub temperature=\\"0\\" scale=\\"Celsius\\"></basetemperature-stub> <!----></basecard-stub></div>"`;
